From b247b734eac30e3cfacf834f15fd733ba8c2ca25 Mon Sep 17 00:00:00 2001
From: Daniel Nachun <daniel.nachun@gmail.com>
Date: Thu, 31 Jul 2025 13:59:04 -0500
Subject: [PATCH] Add-arm64-toolchain

---
 toolchain/linux/BUILD.gn | 95 ++++++++++++++++++++++++++++++++--------
 1 file changed, 76 insertions(+), 19 deletions(-)

diff --git a/toolchain/linux/BUILD.gn b/toolchain/linux/BUILD.gn
index a6e20a0..44dc225 100644
--- a/toolchain/linux/BUILD.gn
+++ b/toolchain/linux/BUILD.gn
@@ -51,6 +51,25 @@ gcc_toolchain("arm64") {
   }
 }
 
+gcc_toolchain("ppc64") {
+  toolprefix = "powerpc64le-conda-linux-gnu-"
+
+  cc = "${toolprefix}gcc"
+  cxx = "${toolprefix}g++"
+
+  ar = "${toolprefix}ar"
+  ld = cxx
+  readelf = "${toolprefix}readelf"
+  nm = "${toolprefix}nm"
+
+  toolchain_args = {
+    current_cpu = "ppc64"
+    current_os = "linux"
+    is_clang = false
+    host_byteorder = "little"
+  }
+}
+
 gcc_toolchain("arm") {
   toolprefix = "arm-linux-gnueabihf-"
 
@@ -151,6 +170,63 @@ clang_v8_toolchain("clang_x64_v8_arm64") {
   }
 }
 
+gcc_toolchain("x64_v8_arm64") {
+  toolprefix = "x86_64-conda-linux-gnu-"
+
+  cc = "${toolprefix}gcc"
+  cxx = "${toolprefix}g++"
+
+  ar = "${toolprefix}ar"
+  ld = cxx
+  readelf = "${toolprefix}readelf"
+  nm = "${toolprefix}nm"
+
+  toolchain_args = {
+    current_cpu = "x64"
+    v8_current_cpu = "arm64"
+    current_os = "linux"
+    is_clang = false
+  }
+}
+
+gcc_toolchain("x64_v8_ppc64") {
+  toolprefix = "x86_64-conda-linux-gnu-"
+
+  cc = "${toolprefix}gcc"
+  cxx = "${toolprefix}g++"
+
+  ar = "${toolprefix}ar"
+  ld = cxx
+  readelf = "${toolprefix}readelf"
+  nm = "${toolprefix}nm"
+
+  toolchain_args = {
+    current_cpu = "x64"
+    v8_current_cpu = "ppc64"
+    current_os = "linux"
+    is_clang = false
+  }
+}
+
+gcc_toolchain("x86_v8_ppc64") {
+  toolprefix = "x86_64-conda-linux-gnu-"
+
+  cc = "${toolprefix}gcc"
+  cxx = "${toolprefix}g++"
+
+  ar = "${toolprefix}ar"
+  ld = cxx
+  readelf = "${toolprefix}readelf"
+  nm = "${toolprefix}nm"
+
+  toolchain_args = {
+    current_cpu = "x64"
+    v8_current_cpu = "ppc64"
+    current_os = "linux"
+    is_clang = false
+  }
+}
+
 clang_v8_toolchain("clang_x64_v8_mips64el") {
   toolchain_args = {
     current_cpu = "x64"
@@ -311,25 +387,6 @@ gcc_toolchain("s390x") {
   }
 }
 
-gcc_toolchain("ppc64") {
-  cc = "gcc"
-  cxx = "g++"
-
-  readelf = "readelf"
-  nm = "nm"
-  ar = "ar"
-  ld = cxx
-
-  toolchain_args = {
-    current_cpu = "ppc64"
-    current_os = "linux"
-
-    # remote execution does not support gcc.
-    use_remoteexec = false
-    is_clang = false
-  }
-}
-
 gcc_toolchain("mips") {
   toolprefix = "mips-linux-gnu-"
 
-- 
2.49.0

