From ffbd92f00d3faedb9dcc18b1a7c6873293b799f0 Mon Sep 17 00:00:00 2001
From: Daniel Nachun <daniel.nachun@gmail.com>
Date: Fri, 1 Aug 2025 12:40:27 -0500
Subject: [PATCH] disable vendored libc++

---
 config/BUILDCONFIG.gn | 43 -------------------------------------------
 1 file changed, 43 deletions(-)

diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index a9d2cae..bef7218 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -557,26 +557,6 @@ foreach(_target_type,
         deps = []
       }
 
-      if (is_clang && (!defined(invoker.use_libcxx_modules) ||
-                       invoker.use_libcxx_modules)) {
-        # This is necessary for Clang modules builds.
-        deps += [
-          "//buildtools/third_party/libc++:std",
-          "//buildtools/third_party/libc++:std_config",
-          "//buildtools/third_party/libc++:std_core",
-          "//buildtools/third_party/libc++:std_ctype_h",
-          "//buildtools/third_party/libc++:std_errno_h",
-          "//buildtools/third_party/libc++:std_fenv_h",
-          "//buildtools/third_party/libc++:std_float_h",
-          "//buildtools/third_party/libc++:std_inttypes_h",
-          "//buildtools/third_party/libc++:std_math_h",
-          "//buildtools/third_party/libc++:std_private_mbstate_t",
-          "//buildtools/third_party/libc++:std_string_h",
-          "//buildtools/third_party/libc++:std_uchar_h",
-          "//buildtools/third_party/libc++:std_wctype_h",
-        ]
-      }
-
       # Consumed by the unsafe-buffers plugin during compile.
       #
       # TODO(crbug.com/326584510): Reclient doesn't respect this variable, see
@@ -661,29 +641,6 @@ foreach(_target_type,
         deps += [ "//build/config:${_target_type}_deps" ]
       }
 
-      if (is_clang && (!defined(invoker.use_libcxx_modules) ||
-                       invoker.use_libcxx_modules)) {
-        # These are necessary for Clang modules builds.
-        deps += [
-          "//buildtools/third_party/libc++:_Builtin_limits",
-          "//buildtools/third_party/libc++:_Builtin_stdarg",
-          "//buildtools/third_party/libc++:_Builtin_stddef",
-          "//buildtools/third_party/libc++:std",
-          "//buildtools/third_party/libc++:std_config",
-          "//buildtools/third_party/libc++:std_core",
-          "//buildtools/third_party/libc++:std_ctype_h",
-          "//buildtools/third_party/libc++:std_errno_h",
-          "//buildtools/third_party/libc++:std_fenv_h",
-          "//buildtools/third_party/libc++:std_float_h",
-          "//buildtools/third_party/libc++:std_inttypes_h",
-          "//buildtools/third_party/libc++:std_math_h",
-          "//buildtools/third_party/libc++:std_private_mbstate_t",
-          "//buildtools/third_party/libc++:std_string_h",
-          "//buildtools/third_party/libc++:std_uchar_h",
-          "//buildtools/third_party/libc++:std_wctype_h",
-        ]
-      }
-
       # On Android, write shared library output file to metadata. We will use
       # this information to, for instance, collect all shared libraries that
       # should be packaged into an APK.
-- 
2.49.0

