--- test/BUILD.gn	2020-07-20 21:29:09.000000000 +0200
+++ test/BUILD.gn	2020-07-20 21:29:39.000000000 +0200
@@ -25,15 +25,6 @@
     "mkgrokdump:mkgrokdump",
     "wasm-api-tests:wasm_api_tests",
   ]
-
-  if (host_os != "mac" || !is_android) {
-    # These items don't compile for Android on Mac.
-    deps += [
-      "cctest:cctest",
-      "cctest:generate-bytecode-expectations",
-      "unittests:unittests",
-    ]
-  }
 }
 
 ###############################################################################
